packages/plugins/users-permissions/server/services/providers-list.js

```diff
@@ -2,472 +2,278 @@
+          return {
+            username,
+            email: body.email,
+          };
         });
-      break;
     }
     case 'cognito': {
       // get the id_token
       const idToken = query.id_token;
       // decode the jwt token
       const tokenPayload = jwt.decode(idToken);
       if (!tokenPayload) {
-        callback(new Error('unable to decode jwt token'));
+        throw new Error('unable to decode jwt token');
       } else {
-        callback(null, {
+        return {
           username: tokenPayload['cognito:username'],
           email: tokenPayload.email,
-        });
+        };
       }
-      break;
     }
     case 'facebook': {
-      const facebook = purest({
-        provider: 'facebook',
-        config: purestConfig,
-      });
+      const facebook = purest({ provider: 'facebook' });
 
-      facebook
-        .query()
-        .get('me?fields=name,email')
+      return facebook
+        .get('me')
         .auth(access_token)
-        .request((err, res, body) => {
-          if (err) {
-            callback(err);
-          } else {
-            callback(null, {
-              username: body.name,
-              email: body.email,
-            });
-          }
-        });
-      break;
+        .qs({ fields: 'name,email' })
+        .request()
+        .then(({ body }) => ({
+          username: body.name,
+          email: body.email,
+        }));
     }
     case 'google': {
-      const google = purest({ provider: 'google', config: purestConfig });
+      const google = purest({ provider: 'google' });
 
-      google
+      return google
         .query('oauth')
         .get('tokeninfo')
         .qs({ access_token })
-        .request((err, res, body) => {
-          if (err) {
-            callback(err);
-          } else {
-            callback(null, {
-              username: body.email.split('@')[0],
-              email: body.email,
-            });
-          }
-        });
-      break;
+        .request()
+        .then(({ body }) => ({
+          username: body.email.split('@')[0],
+          email: body.email,
+        }));
     }
     case 'github': {
       const github = purest({
         provider: 'github',
-        config: purestConfig,
         defaults: {
           headers: {
             'user-agent': 'strapi',
           },
         },
       });
 
-      github
-        .query()
+      return github
         .get('user')
         .auth(access_token)
-        .request((err, res, userbody) => {
-          if (err) {
-            return callback(err);
-          }
-
+        .request()
+        .then(({ body: userbody }) => {
           // This is the public email on the github profile
           if (userbody.email) {
-            return callback(null, {
+            return {
               username: userbody.login,
               email: userbody.email,
-            });
+            };
           }
-
           // Get the email with Github's user/emails API
           github
```

Comment:
We are missing a return here. the promise will resolve undefined instead of the object